org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/FeatureCallCompiler.java

```diff
@@ -1042,6 +1047,28 @@ protected void appendInlineFeatureCall(XAbstractFeatureCall call, ITreeAppendabl
 		if (prevEnd != formatString.length()) {
 			b.append(formatString.substring(prevEnd));
 		}
+		if (inlineCallNeedsParenthesis) {
+			b.append(")");
+		}
+	}
+	
+	protected boolean inlineCallNeedsParenthesis(XAbstractFeatureCall call, String formatString) {
+		String trimmedFormatString = formatString.trim();
+		if (trimmedFormatString.startsWith("(")) {
+			return false;
+		}
+		if (call.eContainer() instanceof XMemberFeatureCall) {
+			XMemberFeatureCall mfc = (XMemberFeatureCall) call.eContainer();
+			JvmAnnotationReference inlineAnnotation = expressionHelper.findInlineAnnotation(mfc);
+			if (inlineAnnotation != null) {
+				return true;
+			}
+			if (mfc.isExtension()) {
+				return false;
+			}
+			return !trimmedFormatString.endsWith(")");
```

Comment:
Should this condition be added above?

Reply:
am not sure about this cause it would catch the "extension" case too

Reply:
I see